mgnify_pipelines_toolkit/analysis/shared/study_summary_generator.py

```diff
@@ -0,0 +1,263 @@
+            res_df = res_df.join(df, how="outer")
+        res_df = res_df.fillna(0)
+
+        res_df.to_csv(
+            f"{output_prefix}_{db_label}_study_summary.tsv",
+            sep="\t",
+            index_label="taxonomy",
+        )
+
+    elif db_label in ASV_TAXDB_LABELS:
+
+        amp_region_dict = defaultdict(list)
+
+        for (
+            run_acc,
+            tax_file_asv_lst,
+        ) in (
+            tax_files.items()
+        ):  # each `tax_file` will be a list containing at most two files (one for each amp_region)
+            for tax_file in tax_file_asv_lst:
+                amp_region = str(tax_file).split("_")[
+                    -5
+                ]  # there are a lot of underscores in these names... but it is consistent
+                amp_region_df = parse_one_tax_file(run_acc, tax_file)
+                amp_region_dict[amp_region].append(amp_region_df)
+
+        for amp_region, amp_region_dfs in amp_region_dict.items():
+            if (
+                len(amp_region_dfs) > 1
+            ):  # Need at least two analyses with this amp_region to bother with the summary
+                amp_res_df = amp_region_dfs[0]
+                for amp_df in amp_region_dfs[1:]:
+                    amp_res_df = amp_res_df.join(amp_df, how="outer")
+                amp_res_df = amp_res_df.fillna(0)
+
+                amp_res_df.to_csv(
+                    f"{output_prefix}_{db_label}_{amp_region}_asv_study_summary.tsv",
+                    sep="\t",
+                    index_label="taxonomy",
+                )
+
+
+def summarise_analyses(runs_df, analyses_dir, output_prefix):
+
+    all_db_labels = TAXDB_LABELS + ASV_TAXDB_LABELS
+    for db_label in all_db_labels:
+
+        tax_files = defaultdict(Path)
+        for i in range(0, len(runs_df)):
+            run_acc = runs_df.loc[i, "run"]
+            tax_file = get_tax_file(run_acc, analyses_dir, db_label)
+
+            if tax_file:
+                tax_files[run_acc] = tax_file
+
+        if (
+            len(tax_files) > 1
+        ):  # If at least two analyses have results from the current DB, generate a study-level summary for it
+            generate_db_summary(db_label, tax_files, output_prefix)
+
+
+def organise_study_summaries(all_study_summaries):
+
+    summaries_dict = defaultdict(list)
+
+    for summary in all_study_summaries:
+        summary_path = Path(summary)
+        summary_filename = summary_path.stem
+
+        temp_lst = summary_filename.split("_")
+        if "asv_study_summary" in summary_filename:
+            summary_db_label = "_".join(
+                temp_lst[1:3]
+            )  # For ASVs we need to include the amp_region in the label
+        else:
+            summary_db_label = temp_lst[
+                1
+            ]  # For closed reference, just the db_label is needed
+
+        summaries_dict[summary_db_label].append(summary_path)
+
+    return summaries_dict
+
+
+def merge_summaries(analyses_dir, output_prefix):
```

Comment:
I'd add type hints here. It is likely that in prefect world we would call this as a library method, directly from python, rather than as CLI. So, we won't go through argparser and therefore type hints are very helpful.
```suggestion
def merge_summaries(analyses_dir: Path, output_prefix: str) -> None:
```